internal/services/nss/nss_test.go

```diff
@@ -37,46 +37,24 @@ func TestNewService(t *testing.T) {
 
 	_ = nss.NewService(context.Background(), m, b)
 }
-
-func TestUserPreCheck(t *testing.T) {
-	tests := map[string]struct {
-		username string
-
-		wantErr bool
-	}{
-		"Return existing user": {username: "user1"},
-
-		"Error if user does not exist in any broker": {username: "does-not-exist", wantErr: true},
-	}
-	for name, tc := range tests {
-		tc := tc
-		t.Run(name, func(t *testing.T) {
-			client := newNSSClient(t, "empty.db.yaml")
-
-			_, err := client.UserPreCheck(context.Background(), &authd.GetByNameRequest{Name: tc.username})
-			if tc.wantErr {
-				require.Error(t, err, "UserPreCheck should return an error but did not")
-				return
-			}
-			require.NoError(t, err, "UserPreCheck should not return an error, but did")
-		})
-	}
-}
-
 func TestGetPasswdByName(t *testing.T) {
 	tests := map[string]struct {
 		username string
 
-		sourceDB string
+		sourceDB       string
+		shouldPreCheck bool
```

Comment:
I'd keep this as a pointer so that we also test the `nil` case.

Reply:
Honestly, I think the best course of action is actually to modify the proto so that the precheck field is not optional. I don't like the idea of a pointer for something that can only be Y or N.

Reply:
Oh, that too... Not sure what was the decision on protocol with other peers.